packages/server-wallet/src/protocols/defunder.ts

```diff
@@ -90,14 +90,18 @@ export class Defunder {
   }
 
   private async ledgerDefunder(channel: Channel, tx: Transaction): Promise<DefunderResult> {
+    const didSubmitTransaction = false;
```

Comment:
What's the purpose of `didSubmitTransaction` if it is a constant with value `false`? That makes it seem like it doesn't belong on `DefunderResult`, as its value appears to contain no information.

e. Ok, I now realize that `ledgerDefunder` should never return `didSubmitTransaction: true`. I wonder if this should be clarified via a code comment.

Reply:
Curious if 3dddf847f helps

Reply:
It's very clarifying! But it does cause me to ask: Why doesn't `Defunder.crank` just return a `boolean`? I'm going to read the surrounding context to try to answer that.

Reply:
Actually, it seems like
- `didSubmitTransaction` is used purely for testing within this PR.
- On `master`, [`didSubmitTransaction`](https://github.com/statechannels/statechannels/blob/3d6270b0c49914ff0f8d367fce340189f4924225/packages/server-wallet/src/protocols/channel-closer.ts#L60) is not used

So, it seems to me like `Defunder.crank` should just return a `boolean`?

Reply:
What about https://github.com/statechannels/statechannels/blob/3d6270b0c49914ff0f8d367fce340189f4924225/packages/server-wallet/src/protocols/defund-channel.ts#L56?

Reply:
Now I'm confused.
1. Why does a `ChannelDefunder` use a `Defunder`? What is the 2nd defunder for?
2. Why doesn't the `ChannelDefunder` just ask if `Defunder.crank` returned `true`?

Reply:
`ChannelDefunder` was written before much thought went into first-class objectives. I don't think we will be using that protocol going forward. To eliminate `ChannelDefunder`, we need to think through how challenges work with first class objectives. It probably doesn't make sense to refactor `ChannelDefunder` and its test. My proposal going forward is (which is what has been implemented):
1. Enable `ChannelDefunder`'s current behavior via `didSubmitTransaction` return value.
2. When `ChannelDefunder` is removed, remove `didSubmitTransactoin`.

Reply:
Sounds good. Should we record that proposal somewhere?

Reply:
That's a good suggestion! https://github.com/statechannels/statechannels/issues/3309